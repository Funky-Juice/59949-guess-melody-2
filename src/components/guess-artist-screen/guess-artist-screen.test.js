import renderer from 'react-test-renderer';
import GuessArtistScreen from './guess-artist-screen';

const mockQuestion = {
  type: `artist`,
  song: {
    artist: `Ozzy Osbourne`,
    src: `https://dl1.ru-music.xn--41a.ws/mp3/1369.mp3`
  },
  answers: [
    {
      picture: `https://upload.wikimedia.org/wikipedia/commons/thumb/7/7b/Till_Lindemann_-_2017287140953_2017-10-14_Buchmesse.jpg/800px-Till_Lindemann_-_2017287140953_2017-10-14_Buchmesse.jpg`,
      artist: `Till Lindemann`
    },
    {
      picture: `https://upload.wikimedia.org/wikipedia/commons/f/fd/JamesHetfield2012.jpg`,
      artist: `James Hetfield`
    },
    {
      picture: `https://upload.wikimedia.org/wikipedia/commons/7/7b/Ozzy_on_tour_in_Japan.jpg`,
      artist: `Ozzy Osbourne`
    }
  ]
};


it(`GuessArtistScreen correctly renders`, () => {
  const tree = renderer
    .create(<GuessArtistScreen question={mockQuestion} time={0} errors={0} screenIndex={0} onAnswer={() => {}}/>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
