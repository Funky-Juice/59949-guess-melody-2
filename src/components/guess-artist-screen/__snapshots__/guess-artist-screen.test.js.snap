// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuessArtistScreen correctly renders 1`] = `
<article
  id="game-artist"
>
  <section
    className="game game--artist"
  >
    <header
      className="game__header"
    >
      <a
        className="game__back"
        href="#"
      >
        <span
          className="visually-hidden"
        >
          Сыграть ещё раз
        </span>
        <img
          alt="Угадай мелодию"
          className="game__logo"
          src="img/melody-logo-ginger.png"
        />
      </a>
      <svg
        className="timer"
        viewBox="0 0 780 780"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          className="timer__line"
          cx="390"
          cy="390"
          r="370"
          style={
            Object {
              "filter": "url(#blur)",
              "transform": "rotate(-90deg) scaleY(-1)",
              "transformOrigin": "center",
            }
          }
        />
      </svg>
      <div
        className="timer__value"
        xmlns="http://www.w3.org/1999/xhtml"
      >
        <span
          className="timer__mins"
        >
          0
        </span>
        <span
          className="timer__dots"
        >
          :
        </span>
        <span
          className="timer__secs"
        >
          00
        </span>
      </div>
      <div
        className="game__mistakes"
      />
    </header>
    <section
      className="game__screen"
    >
      <h2
        className="game__title"
      >
        Кто исполняет эту песню?
      </h2>
      <div
        className="game__track"
      >
        <div
          className="track"
        >
          <button
            className="track__button track__button--play"
            type="button"
          />
          <div
            className="track__status"
          >
            <audio
              controls={true}
              src="https://dl1.ru-music.xn--41a.ws/mp3/1369.mp3"
            />
          </div>
        </div>
      </div>
      <form
        className="game__artist"
      >
        <div
          className="artist"
        >
          <input
            className="artist__input visually-hidden"
            id="answer-0"
            name="answer"
            onChange={[Function]}
            type="radio"
            value="Till Lindemann"
          />
          <label
            className="artist__name"
            htmlFor="answer-0"
          >
            <img
              alt="Till Lindemann"
              className="artist__picture"
              src="https://upload.wikimedia.org/wikipedia/commons/thumb/7/7b/Till_Lindemann_-_2017287140953_2017-10-14_Buchmesse.jpg/800px-Till_Lindemann_-_2017287140953_2017-10-14_Buchmesse.jpg"
            />
            Till Lindemann
          </label>
        </div>
        <div
          className="artist"
        >
          <input
            className="artist__input visually-hidden"
            id="answer-1"
            name="answer"
            onChange={[Function]}
            type="radio"
            value="James Hetfield"
          />
          <label
            className="artist__name"
            htmlFor="answer-1"
          >
            <img
              alt="James Hetfield"
              className="artist__picture"
              src="https://upload.wikimedia.org/wikipedia/commons/f/fd/JamesHetfield2012.jpg"
            />
            James Hetfield
          </label>
        </div>
        <div
          className="artist"
        >
          <input
            className="artist__input visually-hidden"
            id="answer-2"
            name="answer"
            onChange={[Function]}
            type="radio"
            value="Ozzy Osbourne"
          />
          <label
            className="artist__name"
            htmlFor="answer-2"
          >
            <img
              alt="Ozzy Osbourne"
              className="artist__picture"
              src="https://upload.wikimedia.org/wikipedia/commons/7/7b/Ozzy_on_tour_in_Japan.jpg"
            />
            Ozzy Osbourne
          </label>
        </div>
      </form>
    </section>
  </section>
</article>
`;
